engine/render/renderer/swapchain/Swapchain.h

```diff
@@ -276,9 +278,8 @@ class SwapChain
     static SwapChain* instance;
 
     // frame buffers and renderpasses
-    // TODO: Refactor this into a HeapArray
-    VkFramebuffer* swapChainFrameBuffers {VK_NULL_HANDLE};
     RenderPass renderPass;
+    Framebuffer framebuffers;
```

Comment:
The terminology here is confusing. Does the class represent a single frame buffer with many indices, or multiple buffers?

Reply:
It represents a framebuffer, but we need to store multiple framebuffers for every frame to avoid synchronisation issues. We essentially double buffer these by default. I haven't found a use case for having an individual framebuffer yet, so we're going to assume that we'll always want to have one frambuffer per frame.